content/blog/2021/release-0.4.0.md

```diff
@@ -0,0 +1,110 @@
+---
+title: "Release v0.4.0"
+description: "Announcing the v0.4.0 release of BDK"
+author: "Alekos Filini"
+date: "2021-02-17"
+tags: ["rust", "release"]
+hidden: true
+draft: false
+---
+
+A new release of BDK is out: the [`v0.4.0`] release brings updated dependencies, more sanity checks and an overhauled API to build transactions.
+
+You can find the full [v0.4.0 changelog][changelog] on GitHub.
+
+## What's new in v0.4.0
+
+Below are some highlights of the new improved APIs coming with this release:
+
+### A new API to build transaction
+
+The star of this release is the new API designed and implemented by [@llfourn] that brings much more flexibility to the way we create transactions: originally the process of making a transaction involved the creation of a `TxBuilder` which was used
+to configure *how* the wallet should build the transaction. Things like which outputs to create, what `nLockTime` to use, which UTXOs to spend, and much more.
+
+Once fully configured, this builder was then given to the `Wallet` itself in a `Wallet::create_tx()` or `Wallet::bump_fee()` call: the `Wallet` would try to follow the *instructions* given by the builder, but in
+case of conflicting or straight-up wrong options it would have to fail and force the user to start over.
+
+The new API maintains the concept of a *builder*, but it changes the way it's created so that it always contains a reference to the main `Wallet` instance. What this means is that most checks can now be performed right
+when something is added to the builder, not at the end, allowing the user to recover from errors instead of having to start over.
+
+This also opens the door to even more improvements and additions, such as a way to [spend foreign utxos][foreign_utxo] in a transaction, or even a way to [bump the fees of multiple transactions at once][bump_fee_batched] by batching them together, which
+saves a bit of space and money.
+
+```rust
+let send_to = wallet.get_new_address()?;
+let (psbt, details) = {
+    let mut builder = wallet.build_tx();
+    builder
+        .add_recipient(send_to.script_pubkey(), 50_000)
+        .enable_rbf()
+        .do_not_spend_change()
+        .fee_rate(FeeRate::from_sat_per_vb(5.0));
+    builder.finish()?
+};
+```
+
+### Upgraded dependencies
+
+This release also brings many updates to our dependencies, including:
+
+- `bitcoin` to `v0.26`
+- `miniscript` to `v5.1`
+- `electrum-client` to `v0.6`
+- `tokio` to `v1`
+- `reqwest` to `v0.11`
+- `cc` to `>= v1.0.64`
+
+### Compact Filters example
+
+Thanks to the upgrade to `bitcoin v0.26` all the issues related to new networking messages in the P2P Bitcoin network has been fixed, which means that we can finally use our (experimental) compact filters `Blockchain` with
```

Comment:
small grammar fix.. should read "all the issues related to new networking messages in the P2P Bitcoin network _have_ been fixed, ..."